apps/GaussianCurvature/gaussian_curvature_ref.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#include <vector>
+#include "rxmesh/util/report.h"
+
+const double PI = 3.1415926535897932384626433832795028841971693993751058209;
+
+template <typename T>
+inline void gaussian_curvature_ref(const std::vector<std::vector<uint32_t>>& Faces,
+                              const std::vector<std::vector<T>>&        Verts,
+                              std::vector<T>& gaussian_curvature)
+{
+    uint32_t num_vertices = Verts.size();
+    uint32_t num_faces    = Faces.size();
+
+    std::vector<T> region_mixed(gaussian_curvature.size());
+
+    memset((void*)gaussian_curvature.data(), 2 * PI, gaussian_curvature.size() * sizeof(T));
```

Comment:
I think `memset` is not what you want you use here since it fills each byte in `gaussian_curvature.data()` with `2*PI`. Instead, you want to fill each 4 bytes (if `T` is float) with `2*PI`. So, the right thing to use is [`std::fill`](https://en.cppreference.com/w/cpp/algorithm/fill)

Reply:
Yes. That's what I did to solve this.